specification/sbi_atee.adoc

```diff
@@ -260,13 +260,19 @@ is that the host will service a subsequent page-fault that results from a TVM-ac
 TVMs can choose to yield access to confidential memory at runtime and request shared (non-confidential) memory.
 The TVM must communicate its request to the host to convert confidential to non-confidential and vice-versa 
 explicitly via the `sbi_tee_guest_share_memory_region()` and `sbi_tee_guest_unshare_memory_region()`. This request
-results in an exit to the TSM which enforces the security properties on the mapping and exits to the VMM host to 
-enforce TLB invalidation. The host must complete a TVM TLB invalidation sequence, initiated by `tee_host_tvm_initiate_fence()`,
-to complete the assignment change. The calling TVM vCPU is considered blocked until the assignment change is completed.
-Attempts to run it with `sbi_tee_host_run_tvm_vcpu()` will fail. Any guest page faults taken by other TVM vCPUs in this
-region before the completion of the assignment change are considered fatal. The host may not insert any pages in the region
-before the completion of the assignment change. Upon completion, the host may either immediately add non-confidential pages
-or may insert the pages on faults during TVM access into the region using `sbi_tee_host_add_tvm_shared_pages()`.
+results in an exit to the TSM which enforces the security properties on the mapping and exits to the VMM host.
+If the region of address space is populated, the host must invalidate and block the pages using 
+`sbi_tee_host_tvm_block_pages()` followed by a TVM TLB invalidation sequence, initiated by
+`tee_host_tvm_initiate_fence()`, in order to complete the invalidation. Then the host must remove the pages from
+TVM page-table using `sbi_tee_host_tvm_remove_pages()`. Upon completion, the host may reclaim the confidential
+pages that were previously mapped in the region using `tee_host_tsm_reclaim_pages()`. The host must then return
+and insert shared pages into the region using `tee_host_tvm_add_shared_pages()` on a page-fault when TVM tries to access
+the region. If the region of address space is unpopulated, the host can immediately return and insert shared pages into
+the region using `tee_host_tvm_add_shared_pages()` on the next page-fault without block, fence and remove sequence.
+
+The calling TVM vCPU is considered blocked until the assignment-change is completed. Attempts to run it
+with `sbi_tee_host_run_tvm_vcpu()` will fail. Any guest page faults taken by other TVM vCPUs in the
+blocked pages continue to be reported to the host.
 
```

Comment:
I feel some text about the reasoning behind host initiated block/fence/remove flow would serve better rather than just explaining the flow which the function description does anyways.

Reply:
Commits removed. I will be opening a new PR for that after reworking the share/unshare parts.